File: RecipeRandomizer.Web/ClientApp/src/app/models/ingredient.ts
import {QuantityUnit} from './nomenclature/quantityUnit';
import {Recipe} from './recipe';

export class Ingredient {
    id: number;
    quantityUnitId: number;
    recipeId: number;
    name: string;
    quantity: number;

    quantityUnit: QuantityUnit;
    recipe: Recipe;
}

File: RecipeRandomizer.Data/Entities/Ingredient.cs
using RecipeRandomizer.Data.Entities.Nomenclature;

namespace RecipeRandomizer.Data.Entities
{
    public class Ingredient
    {
        public int Id { get; set; }
        public int QuantityUnitId { get; set; }
        public int RecipeId { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }

        public virtual QuantityUnit QuantityUnit { get; set; }
        public virtual Recipe Recipe { get; set; }
    }
}
